### PR TITLE
Sandbox with perps

### PR DESCRIPTION
**Pull request by noisekit**
Initial version.

- [x] Commented out pool config TODO: figure out why it fails with market does not exist (for any of the deployed ones) - cc @sunnyvempati

- [x] Still have pyth even though we use constant oracles. TODO: figure out settlement strategy and avoid pyth -- this will have to stay for now

- [x] susd in wallet. 
  I did check as test user with
  ```sh
  yarn cannon run synthetix-sandbox-with-perps:1@main --impersonate 0x70997970C51812dc3A010C7d01b50e0d17dc79C8
  ```
  ~~And looks like `BOX` balance is correct. But it did have 0 balance from `USDProxy`, though user borrowed 300 usd.~~ UPD: needed to withdraw! All good now

- [x] check the USD thing maybe something is missing? UPD: perps interactions coming in following PRs. Basics are working.

### CODE REVIEW

File: sandboxes/with-perps/cannonfile.toml [hunk 1]
```diff
@@ -0,0 +1,622 @@
...
+args = ["<%= extras.spartan_council_pool_id %>"]
+
+[invoke.configure_spartan_council_pool]
+target = ["synthetix.CoreProxy"]
+fromCall.func = "getPoolOwner"
+fromCall.args = ["<%= extras.spartan_council_pool_id %>"]
+func = "setPoolConfiguration"
+args = [
+    "<%= extras.spartan_council_pool_id %>",
+    # TODO: for some reason all of these markets fail here with "Market does not exist" error. Need to figure it out
```
Comment (noisekit) [inline sandboxes/with-perps/cannonfile.toml:498]: This issue is reproduced here https://github.com/Synthetixio/synthetix-sandbox/pull/7
Something to do with `salt` apparently.

**Comment by kaleb-keny** [comment]
this is a pretty handy tool and would like to receive updates once it's ready to test it out

File: sandboxes/with-perps/cannonfile.toml (hunk 1 above)
Comment (noisekit) [reply sandboxes/with-perps/cannonfile.toml]: This is solved by importing external packages instead of provisioning them

**Review by dbeal-eth — APPROVED** [review]

File: sandboxes/with-perps/cannonfile.consumer.toml [hunk 2]
```diff
@@ -0,0 +1,104 @@
+name = "perps-sandbox-consumer-example"
+version = "1"
+description = "Consumer of Perps Sandbox"
+
+#
+#
+# Provision Synthetix Sanbox with perps
+#
+#
+[import.perps_sanbox]
```
Comment (dbeal-eth) [inline sandboxes/with-perps/cannonfile.consumer.toml]: `perps_sanbox`?

File: sandboxes/with-perps/cannonfile.consumer.toml [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+]
+depends = ["invoke.test_user_approves_10k_box_for_CoreProxy_to_spend"]
+
+[invoke.test_user_delegates_900_box]
+target = ["perps_sanbox.synthetix.CoreProxy"]
+from = "<%= settings.test_user %>"
+func = "delegateCollateral"
+args = [
+    # examples of how to debug internal values line by line
+    "<%= /* console.log(extras) || */ extras.test_user_account_id %>",
```
Comment (dbeal-eth) [inline sandboxes/with-perps/cannonfile.consumer.toml:61]: i actually ditnt know this was possible

File: sandboxes/with-perps/cannonfile.consumer.toml (hunk 2 above)
Comment (noisekit) [reply sandboxes/with-perps/cannonfile.consumer.toml]: Yeah already fixed.

File: sandboxes/with-perps/cannonfile.consumer.toml (hunk 3 above)
Comment (noisekit) [reply sandboxes/with-perps/cannonfile.consumer.toml:61]: I wanted to output things into files but `require` was not available ;)

File: sandboxes/with-perps/cannonfile.consumer.toml [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+depends = ["invoke.test_user_borrows_300_susd"]
+
+#
+#
+#
+# Interacting with Perps market
+#
+#
+#
+[invoke.test_user_creates_perps_account]
```
Comment (noisekit) [inline sandboxes/with-perps/cannonfile.consumer.toml:104]: image

File: sandboxes/with-perps/cannonfile.consumer.toml [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+[invoke.approve_300_susd_for_PerpsMarketProxy_to_spend]
+target = ["perps_sandbox.synthetix.USDProxy"]
+from = "<%= settings.test_user %>"
+func = "approve"
+args = [
+    "<%= imports.perps_sandbox.imports.perps_factory.contracts.PerpsMarketProxy.address %>",
+    "<%= parseEther('300') %>",
+]
+
+[invoke.deposit_300_susd_to_perps]
```
Comment (noisekit) [inline sandboxes/with-perps/cannonfile.consumer.toml:123]: image

**Pull request merged**